plugins/BEdita/Core/src/Model/Table/UsersTable.php

```diff
@@ -392,7 +392,14 @@ protected function anonymizeUser(EntityInterface $entity)
         $entity->set('deleted', true);
         $entity->set('locked', true);
 
-        $this->ExternalAuth->deleteAll([$this->ExternalAuth->aliasField('user_id') => $entity->get('id')]);
+        $condition = [$this->ExternalAuth->aliasField('user_id') => $entity->get('id')];
+        $extAuthList = $this->ExternalAuth->find()->where($condition)->toList();
+        array_walk(
+            $extAuthList,
+            function ($item) {
+                $this->ExternalAuth->deleteOrFail($item);
+            }
+        );
```

Comment:
I'd leave the marshaller do the loading, so we don't need to explicitly write conditions.

```suggestion
        $this->loadInto($entity, ['ExternalAuth']);
        array_walk(
            $entity->external_auths,
            function ($item) {
                $this->ExternalAuth->deleteOrFail($item);
            }
        );
```